"optimize" is not exported from module "sqlglot.optimizer"

```
"optimize" is not exported from module "sqlglot.optimizer"
  Import from "sqlglot.optimizer.optimizer" instead
```

for:
```python
from sqlglot.optimizer import optimize
```


![CleanShot 2024-04-26 at 15 01 41@2x](https://github.com/tobymao/sqlglot/assets/125105/e32ea679-f980-4acd-ba81-35b5e37112f1)

I think it needs to be added to `__init__.py`

mypy in strict mode will error too:
```
❯ .venv/bin/mypy --strict fakesnow/test.py
fakesnow/test.py:1: error: Module "sqlglot.optimizer" does not explicitly export attribute "optimize"  [attr-defined]
```

#1203 is a similar issue addressed in the same way.

As background context, pyright (and therefore pylance) considers these symbols private based on [these rules](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface) derived from:

- [PEP 008](https://www.python.org/dev/peps/pep-0008/#id50) states modules should explicitly declare the names in their public API using the `__all__` attribute .
- [PEP 484](https://www.python.org/dev/peps/pep-0484/#stub-files) also mentions that imported modules are not considered exported unless they use the `import ... as ...` syntax or equivalent.

--- a/sqlglot/optimizer/__init__.py
+++ b/sqlglot/optimizer/__init__.py
@@ -1,11 +1,11 @@
 # ruff: noqa: F401
 
-from sqlglot.optimizer.optimizer import RULES, optimize
+from sqlglot.optimizer.optimizer import RULES as RULES, optimize as optimize
 from sqlglot.optimizer.scope import (
-    Scope,
-    build_scope,
-    find_all_in_scope,
-    find_in_scope,
-    traverse_scope,
-    walk_in_scope,
+    Scope as Scope,
+    build_scope as build_scope,
+    find_all_in_scope as find_all_in_scope,
+    find_in_scope as find_in_scope,
+    traverse_scope as traverse_scope,
+    walk_in_scope as walk_in_scope,
 )

